pixl_core/src/core/uploader/_dicomweb.py

```diff
@@ -0,0 +1,121 @@
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+"""Uploader subclass for FTPS uploads."""
+
+from __future__ import annotations
+
+import json
+from typing import Optional
+
+import requests
+from decouple import config  # type: ignore [import-untyped]
+from loguru import logger
+
+from core.uploader.base import Uploader
+
+
+class DicomWebUploader(Uploader):
+    """Upload strategy for a DicomWeb server."""
+
+    def __init__(self, project_slug: str, keyvault_alias: Optional[str]) -> None:
+        """Create instance of parent class"""
+        super().__init__(project_slug, keyvault_alias)
+
+    def _set_config(self) -> None:
+        # Use the Azure KV alias as prefix if it exists, otherwise use the project name
+        az_prefix = self.keyvault_alias
+        az_prefix = az_prefix if az_prefix else self.project_slug
+
+        self.user = self.keyvault.fetch_secret(f"{az_prefix}--dicomweb--username")
+        self.password = self.keyvault.fetch_secret(f"{az_prefix}--dicomweb--password")
+        self.orthanc_url = config("ORTHANC_URL")
+        self.endpoint_name = self.keyvault.fetch_secret(f"{az_prefix}--dicomweb--url")
+        self.url = self.orthanc_url + "/dicom-web/servers/" + self.endpoint_name
+
+    def upload_dicom_image(self) -> None:
+        msg = "Currently not implemented. Use `send_via_stow()` instead."
+        raise NotImplementedError(msg)
+
+    def send_via_stow(self, resource_id: str) -> requests.Response:
+        """Upload a Dicom resource to the DicomWeb server from within Orthanc."""
+        if not self._check_dicomweb_server():
+            logger.info("Creating new DICOMWeb credentials")
+            self._setup_dicomweb_credentials()
+
+        headers = {"content-type": "application/dicom", "accept": "application/dicom+json"}
+        payload = {"Resources": [resource_id], "Synchronous": False}
+        success_code = 200
+
+        try:
+            response = requests.post(
+                self.url + "/stow",
+                auth=(self.user, self.password),
+                headers=headers,
+                data=json.dumps(payload),
+                timeout=30,
+            )
+        except requests.exceptions.RequestException:
+            logger.error("Failed to send via stow")
+            raise
+        else:
+            if response.status_code != success_code:
+                logger.error("Upload via stow unsuccesful")
+                raise RuntimeError
```

Comment:
is there a reason we're not using raise_for_status here?

Reply:
Errr nope! Will change :)